diff --git a/arch/riscv/boot/dts/starfive/jh7110.dtsi b/arch/riscv/boot/dts/starfive/jh7110.dtsi
index b6a6386b9551..2ee85808f816 100644
--- a/arch/riscv/boot/dts/starfive/jh7110.dtsi
+++ b/arch/riscv/boot/dts/starfive/jh7110.dtsi
@@ -372,13 +372,16 @@
 					#size-cells = <1>;
 
 					spl@0 {
-						reg = <0x0 0x40000>;
+						reg = <0x0 0x80000>;
 					};
 					uboot@100000 {
-						reg = <0x100000 0x300000>;
+						reg = <0x100000 0x400000>;
 					};
-					data@f00000 {
-						reg = <0xf00000 0x100000>;
+					uboot-env@f0000 {
+						reg = <0xf0000 0x10000>;
+					};
+					reserved-data@600000 {
+						reg = <0x600000 0x1000000>;
 					};
 				};
 			};
