diff -r 6e87e8960bc2 arch/riscv/Makefile
--- a/arch/riscv/Makefile	Sun May 19 00:25:36 2024 +0700
+++ b/arch/riscv/Makefile	Sun May 19 01:01:27 2024 +0700
@@ -61,6 +61,10 @@
 riscv-march-$(CONFIG_RISCV_ISA_C)	:= $(riscv-march-y)c
 riscv-march-$(CONFIG_RISCV_ISA_V)	:= $(riscv-march-y)v
 
+# Dirty hack for "b" extension on Sifive-U74
+toolchain-has-zba-zbb := $(call cc-option-yn, -march=$(riscv-march-y)_zba_zbb)
+riscv-march-$(toolchain-has-zba-zbb) := $(riscv-march-y)_zba_zbb
+
 ifdef CONFIG_TOOLCHAIN_NEEDS_OLD_ISA_SPEC
 KBUILD_CFLAGS += -Wa,-misa-spec=2.2
 KBUILD_AFLAGS += -Wa,-misa-spec=2.2
